### PR TITLE
Amaterasu 8

### PR DESCRIPTION
**Pull request by roadan**

### CODE REVIEW

File: leader/src/main/scala/org/apache/amaterasu/leader/yarn/ApplicationMaster.scala [hunk 1]
```diff
@@ -205,20 +206,20 @@ class ApplicationMaster extends Logging with AMRMClientAsync.CallbackHandler {
...
-
-    val session = conn.createSession(false, Session.AUTO_ACKNOWLEDGE)
-    //TODO: move to a const in common
-    val destination = session.createTopic("JOB.REPORT")
-
-    val consumer = session.createConsumer(destination)
-    consumer.setMessageListener(new ActiveReportListener)
-
-  }
+//  private def setupMessaging(jobId: String): Unit = {
```
Comment (nadav-har-tzvi) [inline leader/src/main/scala/org/apache/amaterasu/leader/yarn/ApplicationMaster.scala]: Is this piece of commented code needed or can it simply be removed?

File: common/src/main/scala/org/apache/amaterasu/common/execution/actions/Notifier.scala [hunk 2]
```diff
@@ -14,43 +14,43 @@
...
-  val code: NotificationLevel.Value = Value("code")
-  val none: NotificationLevel.Value = Value("none")
-
-}
-
-case class Notification(@JsonProperty("line") line: String,
-                        @JsonProperty("msg") msg: String,
-                        @JsonProperty("notType") notType: NotificationType,
-                        @JsonProperty("notLevel") notLevel: NotificationLevel)
+//package org.apache.amaterasu.common.execution.actions
```
[the remark this answers is not in the document]
Comment (roadan) [reply common/src/main/scala/org/apache/amaterasu/common/execution/actions/Notifier.scala]: You are right, it was re-written in kotlin should be deleted

File: common/build.gradle [hunk 3]
```diff
@@ -46,15 +46,21 @@ dependencies {
...
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.9.4'
 
-
     compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
     compile "org.jetbrains.kotlin:kotlin-reflect"
 
     // currently we have to use this specific mesos version to prevent from
     // clashing with spark
-    compile('org.apache.mesos:mesos:0.22.2:shaded-protobuf') {
+    //compile('org.apache.mesos:mesos:0.28.0:shaded-protobuf') {
```
[the remark this answers is not in the document]
Comment (roadan) [reply common/build.gradle]: Shouldn't have, I've fixed it

File: common/src/main/kotlin/org/apache/amaterasu/common/utils/ArtifactUtil.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    init {
+        val jarFile = File(this.javaClass.protectionDomain.codeSource.location.path)
+        val amaHome = File(jarFile.parent).parent
+        repo = File("$amaHome/dist/$jobId")
+    }
+
+    private val remoteRepos: MutableList<RemoteRepository> = mutableListOf()
+
+    init {
+        addRepos(repos)
```
[the remark this answers is not in the document]
Comment (roadan) [reply common/src/main/kotlin/org/apache/amaterasu/common/utils/ArtifactUtil.kt]: Yes, done

File: frameworks/spark/dispatcher/src/main/scala/org/apache/amaterasu/frameworks/spark/dispatcher/runners/providers/SparkSubmitScalaRunnerProvider.scala [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+class SparkSubmitScalaRunnerProvider extends RunnerSetupProvider {
+
+  private var conf: ClusterConfig = _
+  val jarFile = new File(this.getClass.getProtectionDomain.getCodeSource.getLocation.getPath)
+  val amaDist = new File (s"${new File(jarFile.getParent).getParent}/dist")
+
+  override def getCommand(jobId: String, actionData: ActionData, env: String, executorId: String, callbackAddress: String): String = {
+    val util = new ArtifactUtil(List(actionData.repo).asJava, jobId)
+    val classParam = if (actionData.getHasArtifact)  s" --class ${actionData.entryClass}" else ""
+    s"spark-2.2.1-bin-hadoop2.7/bin/spark-submit $classParam ${util.getLocalArtifacts(actionData.getArtifact).get(0).getName} --deploy-mode client --jars spark-runtime-${conf.version}.jar >&1"
```
[the remark this answers is not in the document]
Comment (roadan) [reply frameworks/spark/dispatcher/src/main/scala/org/apache/amaterasu/frameworks/spark/dispatcher/runners/providers/SparkSubmitScalaRunnerProvider.scala]: Yes, this actually goes beyond what was originally in this PR, I've fixed it now

File: leader-common/src/main/scala/org/apache/amaterasu/leader/common/utilities/DataLoader.scala [hunk 6]
```diff
@@ -48,7 +48,12 @@ object DataLoader extends Logging {
 
   def getTaskData(actionData: ActionData, env: String): TaskData = {
     val srcFile = actionData.getSrc
-    val src = Source.fromFile(s"repo/src/$srcFile").mkString
+    var src = ""
```
[the remark this answers is not in the document]
Comment (nadav-har-tzvi) [reply leader-common/src/main/scala/org/apache/amaterasu/leader/common/utilities/DataLoader.scala]: @roadan again, does it make sense, if not, do we defer dealing with this or do you want to do something about this?

[the remark this answers is not in the document]
Comment (roadan) [reply leader-common/src/main/scala/org/apache/amaterasu/leader/common/utilities/DataLoader.scala]: Yes, when we have an artifact there is no src file, this is why we check first

[the remark this answers is not in the document]
Comment (nadav-har-tzvi) [reply leader-common/src/main/scala/org/apache/amaterasu/leader/common/utilities/DataLoader.scala]: ok

**Pull request merged**